### PR TITLE
Refactor: Admin screens to functional components

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- Screens that are from the loaded wallet, but do not deal with transactions, should be refactored to functional components

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: locale/texts.pot
Comment (andreabadesso) [inline locale/texts.pot:1]: You also need to update the other po files, right?

File: src/screens/AddressList.js [hunk 1]
```diff
@@ -72,132 +44,165 @@ class AddressList extends React.Component {
...
+        fetchedAddresses.push(value);
+      }
+
+      setAddresses(fetchedAddresses);
+      setFilteredAddresses(fetchedAddresses);
+      setTotalPages(getTotalPages(fetchedAddresses));
+    }
+
+    fetchAddresses();
+  }, [wallet]);
```
Comment (andreabadesso) [inline src/screens/AddressList.js:75]: Can `wallet` mutate here? 

If it does, should we cancel the async task?

Maybe something like:

```suggestion
  useEffect(() => {
    let isCancelled = false;
    
    /* useEffect callback cannot be async, so we create a function to do the asynchronous work inside it
     * @see https://devtrium.com/posts/async-functions-useeffect#write-the-asynchronous-function-inside-the-useeffect
     */
    const fetchAddresses = async () => {
      const fetchedAddresses = [];
      const iterator = wallet.getAllAddresses();
      for (;;) {
        if (isCancelled) {
          break;
        }
        const addressObj = await iterator.next();
        const { value, done } = addressObj;

        if (done) {
          break;
        }

        fetchedAddresses.push(value);
      }
      if (!cancelled) {
        setAddresses(fetchedAddresses);
        setFilteredAddresses(fetchedAddresses);
        setTotalPages(getTotalPages(fetchedAddresses));
      }
    }

    fetchAddresses();
    
    return () => {
      isCancelled = true;
    };
  }, [wallet]);
```

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/screens/AddressList.js (hunk 1 above)
Comment (tuliomir) [reply src/screens/AddressList.js:75]: Good point!
`wallet` cannot mutate, as that would mean that the user has reset the wallet and reloaded it again without ever leaving the `AddressList` screen ( which is absurd ).

Adding the `wallet` property to the effect dependencies was a way to follow the guideline _[Don't lie to React about dependencies](https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies)_, but in this case it's not really needed.

So I believe the best course of action would be to remove the dependency altogether and just ignore this guideline as it doesn't fit this screen's business rules.

File: locale/texts.pot
Comment (tuliomir) [reply locale/texts.pot:1]: At the moment there are no other `po` files to update, even though there is support for them on the `check_pot` script.
For now, no further action is needed here.

File: src/screens/AddressList.js (hunk 1 above)
Comment (tuliomir) [reply src/screens/AddressList.js]: Fixed on [b4481d4](https://github.com/HathorNetwork/hathor-wallet/pull/475/commits/b4481d4b75fcc676ec66b76ad5be3d8aaa57c9b0)

**Review by alexruzenhack — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**